File: demo/src/pages/TimerDemo/TimerDemo.ts
import {Button, Viewable,Image, main, Font, State,VStack, Timer,Text, Bound} from "@tswift/ui";
@main
export class TimerDemo extends Viewable {
    @State  value = 0.0;
    timer = Timer.publish( 0.5);//.autoconnect()
    
    body = ({timer, value}:Bound<this>) => 
        VStack ({},
            Image( "rectangle.and.pencil.and.ellipsis", value)
                .imageScale('.large')
                .font(Font.system(60)),
            Image( "wifi", value)
                .imageScale('.large')
                .font(Font.system( 60)),
            Image("phone.down.waves.left.and.right", value)
                .imageScale('.large')
                .font(Font.system( 60)),
            Image( "shower.fill",  value)
                .imageScale('.large')
                .font(Font.system( 60))
                        )
        .onReceive(timer, ()=>{
            if (value < .99) {
                this.value = value + 0.25;
            } else {
                this.value = 0.0;
            }
        })
    
}

File: demo/src/pages/TimerDemo/TimerDemo.swift
import SwiftUI

struct TimerDemo : View {
    @State var value = 0.0
    let timer = Timer.publish(every: 0.5, on: .main, in: .common).autoconnect()
    
    var body: some View {
        VStack {
            Image(systemName: "rectangle.and.pencil.and.ellipsis", variableValue: value)
                .imageScale(.large)
                .font(.system(size: 60))
            Image(systemName: "wifi", variableValue: value)
                .imageScale(.large)
                .font(.system(size: 60))
            Image(systemName: "phone.down.waves.left.and.right", variableValue: value)
                .imageScale(.large)
                .font(.system(size: 60))
            Image(systemName: "shower.fill", variableValue: value)
                .imageScale(.large)
                .font(.system(size: 60))
        }
        .onReceive(timer) { _ in
            if value < 1.0 {
                value += 0.25
            } else {
                value = 0.0
            }
        }
    }
}
